conn.go

```diff
@@ -1768,51 +1796,25 @@ func (c *Conn) query(ctx context.Context, statement string, values ...interface{
 	return c.executeQuery(ctx, q)
 }
 
-func (c *Conn) querySystemPeers(ctx context.Context, version cassVersion) *Iter {
+func (c *Conn) querySystem(ctx context.Context, query string) *Iter {
 	usingClause := ""
 	if c.session.control != nil {
 		usingClause = c.session.usingTimeoutClause
 	}
-	var (
-		peerSchema    = "SELECT * FROM system.peers" + usingClause
-		peerV2Schemas = "SELECT * FROM system.peers_v2" + usingClause
-	)
-
-	c.mu.Lock()
-	if isScyllaConn((c)) { // ScyllaDB does not support system.peers_v2
-		c.isSchemaV2 = false
-	}
-
-	isSchemaV2 := c.isSchemaV2
-	c.mu.Unlock()
-
-	if version.AtLeast(4, 0, 0) && isSchemaV2 {
-		// Try "system.peers_v2" and fallback to "system.peers" if it's not found
-		iter := c.query(ctx, peerV2Schemas)
+	queryStmt := query + usingClause
+	return c.query(ctx, queryStmt)
+}
 
-		err := iter.checkErrAndNotFound()
-		if err != nil {
-			if errFrame, ok := err.(errorFrame); ok && errFrame.code == ErrCodeInvalid { // system.peers_v2 not found, try system.peers
-				c.mu.Lock()
-				c.isSchemaV2 = false
-				c.mu.Unlock()
-				return c.query(ctx, peerSchema)
-			} else {
-				return iter
-			}
-		}
-		return iter
-	} else {
-		return c.query(ctx, peerSchema)
+func querySystemPeers(ctx context.Context, c ConnInterface) *Iter {
+	peerSchema := "SELECT * FROM system.peers"
+	if c.getIsSchemaV2() {
+		peerSchema = peerSchema + "_v2"
 	}
+	return c.querySystem(ctx, peerSchema)
```

Comment:
Let's have two const and get rid of this functions.

Reply:
I am not sure about that, that would mean having to check schema version in three places instead of one in this function. Do you think  it is still worth it?

Reply:
It is only 4 lines of code